tests/tests/shader/data_builtins.rs

```diff
@@ -0,0 +1,127 @@
+    let output: [u32; 4] = [0xDD, 0xCC, 0xBB, 0xAA];
+    let unpack_u8 = ShaderTest::new(
+            format!("unpack4xU8({input:x?}) == {output:x?}"),
+            String::from("value: u32"),
+            String::from("let a = unpack4xU8(input.value); output[0] = a[0];output[1] = a[1];output[2] = a[2];output[3] = a[3];"),
+            &[input],
+            &output,
+        );
+    tests.push(unpack_u8);
+
+    tests
+}
+
+#[gpu_test]
+static UNPACK4xU8: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(
+        TestParameters::default()
+            .downlevel_flags(DownlevelFlags::COMPUTE_SHADERS)
+            .limits(Limits::downlevel_defaults()),
+    )
+    .run_async(|ctx| {
+        shader_input_output_test(ctx, InputStorageType::Storage, create_unpack4xU8_test())
+    });
+
+#[allow(non_snake_case)]
+fn create_unpack4xI8_test() -> Vec<ShaderTest> {
+    let mut tests = Vec::with_capacity(2);
+
+    let values = [
+        // regular unpacking
+        (0x11223344, [0x44, 0x33, 0x22, 0x11]),
+        // sign extension
+        (0xFF, [-1, 0, 0, 0]),
+    ];
+
+    for (input, output) in values {
+        let unpack_i8 = ShaderTest::new(
+            format!("unpack4xI8({input:x?}) == {output:x?}"),
+            String::from("value: u32"),
+            String::from("let a = bitcast<vec4<u32>>(unpack4xI8(input.value)); output[0] = a[0];output[1] = a[1];output[2] = a[2];output[3] = a[3];"),
+            &[input],
+            &output,
+        );
+        tests.push(unpack_i8);
+    }
+
+    tests
+}
+
+#[gpu_test]
+static UNPACK4xI8: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(
+        TestParameters::default()
+            .downlevel_flags(DownlevelFlags::COMPUTE_SHADERS)
+            .limits(Limits::downlevel_defaults()),
+    )
+    .run_async(|ctx| {
+        shader_input_output_test(ctx, InputStorageType::Storage, create_unpack4xI8_test())
+    });
+
+#[allow(non_snake_case)]
+fn create_pack4xU8_test() -> Vec<ShaderTest> {
+    let mut tests = Vec::new();
+
+    let input: [u32; 4] = [0xDD, 0xCC, 0xBB, 0xAA];
+    let output: u32 = 0xAABBCCDD;
+    let pack_u8 = ShaderTest::new(
+        format!("pack4xU8({input:x?}) == {output:x?}"),
+        String::from("value: vec4<u32>"),
+        String::from("output[0] = pack4xU8(input.value);"),
+        &input,
+        &[output],
+    );
+    tests.push(pack_u8);
+
+    tests
+}
+
+#[gpu_test]
+static PACK4xU8: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(
+        TestParameters::default()
+            .downlevel_flags(DownlevelFlags::COMPUTE_SHADERS)
+            .limits(Limits::downlevel_defaults()),
+    )
+    .run_async(|ctx| {
+        shader_input_output_test(ctx, InputStorageType::Storage, create_pack4xU8_test())
+    });
+
+#[allow(non_snake_case)]
+fn create_pack4xI8_test() -> Vec<ShaderTest> {
+    let mut tests = Vec::new();
+
+    let input: [u32; 4] = [0xDD, 0xCC, 0xBB, 0xAA];
```

Comment:
Might test something about sign behavior here?

Reply:
Since the packing operation is essentailly just copying the last 8 bits of the number signess of a value shouldn't matter. But still, more tests could never hurt, so, done!

Reply:
Love the attitude, thanks!